test/scales/scaleTests.ts

```diff
@@ -1,58 +1,184 @@
+      });
     });
 
-    it("quantitaveScale gives the minimum and maxiumum when the domain is numeric", () => {
-      let values = [1, 3, 2, 1];
-      let scale = new Plottable.QuantitativeScale();
-      let computedExtent = scale.extentOfValues(values);
+    describe("included values", () => {
+      it("adding or removing an IncludedValuesProvider returns the Scale", () => {
+        const scale = new MockScale<number, number>();
+        const valueProvider = (): any[] => [];
+        assert.strictEqual(scale.addIncludedValuesProvider(valueProvider), scale, "adding the provider returns the Scale");
+        assert.strictEqual(scale.removeIncludedValuesProvider(valueProvider), scale, "removing the provider returns the Scale");
+      });
+
+      it("combines all values from its IncludedValuesProviders", () => {
+        const scale = new MockScale();
+        const provider1Value = { value: 1 };
+        const provider1 = (passedScale: Plottable.Scale<any, any>) => {
+          assert.strictEqual(passedScale, scale, "was passed the scale it was attached to");
+          return [provider1Value];
+        };
+        const provider2Value = { value: 2 };
+        const provider2 = () => [provider2Value];
+        scale.addIncludedValuesProvider(provider1);
+        scale.addIncludedValuesProvider(provider2);
+
+        const expectedIncludedValues = [provider1Value, provider2Value];
+        assert.deepEqual(scale.includedValues(), expectedIncludedValues, "combines all included values into a single array");
+      });
+
+      it("does not include values from removed IncludedValuesProviders", () => {
+        const scale = new MockScale();
+        const provider1Value = { value: 1 };
+        const provider1 = () => [provider1Value];
+        const provider2Value = { value: 2 };
+        const provider2 = () => [provider2Value];
+
+        scale.addIncludedValuesProvider(provider1);
+        scale.addIncludedValuesProvider(provider2);
+        scale.removeIncludedValuesProvider(provider2);
 
-      assert.deepEqual(computedExtent, [1, 3], "the extent is the miminum and the maximum value in the domain");
+        const expectedIncludedValues = [provider1Value];
+        assert.deepEqual(scale.includedValues(), expectedIncludedValues, "includes only values from remaining providers");
+      });
+
+      it("won't add the same IncludedValuesProvider twice", () => {
+        const scale = new MockScale();
+        const providerValue = { value: 1 };
+        const provider = () => [providerValue];
+
+        scale.addIncludedValuesProvider(provider);
+        scale.addIncludedValuesProvider(provider);
+
+        const expectedIncludedValues = [providerValue];
+        assert.deepEqual(scale.includedValues(), expectedIncludedValues, "only includes the value once");
+      });
     });
 
+    describe("autodomaining", () => {
+      it("overrides a set domain if explicitly autodomained", () => {
+        const expectedDomain = [{}];
```

Comment:
I wanted an object since two objects can't be strict equal to each other, so the guards against a case where the logic just happens to produce the correct values. But I think this is too confusing, so I'll change it to just declare the expected auto and explicitly set values.